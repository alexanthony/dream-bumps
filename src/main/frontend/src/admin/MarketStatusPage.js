// @flow
import React, {PropTypes} from 'react';
import { connect } from 'react-redux';
import {Card, CardText} from 'material-ui/Card';
import {Grid, Cell} from 'material-grid/dist';
import RaisedButton from 'material-ui/RaisedButton';
import Timestamp from 'react-timestamp';

import { updateMarketStatus } from '../status/actions';
import { getCurrentStatus } from '../rootReducer';

class MarketStatusPage extends React.Component {
  onToggleStatusClicked() {
    this.props.updateMarketStatus({open: !this.props.status.open, day: this.props.status.day});
  }

  onNextDayClicked() {
    this.props.updateMarketStatus({open: this.props.status.open, day: this.props.status.day+1});
  }

  render() {
    return (
      <Grid>
        <Cell col={6}>
          <Card>
            <CardText>
              <h3>The market is currently </h3>
              <h1>{this.props.status.open ? 'Open': 'Closed'}</h1>
              <RaisedButton
                primary
                label={this.props.status.open ? 'Close Market':'Open Market'}
                onClick={this.onToggleStatusClicked.bind(this)}
              />
            </CardText>
          </Card>
        </Cell>
        <Cell col={6}>
          <Card>
            <CardText>
              <h3>The market is currently showing</h3>
              <h1>{this.props.status.day >= 4 ? 'Final Finish Order' : `Start Order for Day ${this.props.status.day+1}`}</h1>
              <RaisedButton
                primary
                disabled={this.props.day >= 4}
                label="Next Day"
                onClick={this.onNextDayClicked.bind(this)}
              />
              <p>This cannot be reversed easily. This will also crystallise the dividends for crews. Make sure you have entered all bumps before this.</p>
            </CardText>
          </Card>
        </Cell>
        <Cell col={6}>
          <Card>
            <CardText>
              <p>Market status last updated: <Timestamp time={this.props.status.dateTime} /></p>
            </CardText>
          </Card>
        </Cell>
      </Grid>
    );
  }
}

MarketStatusPage.propTypes = {
  status: PropTypes.object,
  updateMarketStatus: PropTypes.func,
  day: PropTypes.number
};

const mapStateToProps: Object = (state: Object) => ({
  status: getCurrentStatus(state)
});

export default connect(mapStateToProps, { updateMarketStatus })(MarketStatusPage);
