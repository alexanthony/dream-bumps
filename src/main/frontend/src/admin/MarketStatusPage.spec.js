import React from 'react';
import { shallow } from 'enzyme';
import { MarketStatusPage } from './MarketStatusPage';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const updateMarketStatus = jest.fn();
  const component = shallow(
    <MarketStatusPage
      updateMarketStatus={updateMarketStatus}
      status={{day: 3, open: true, dateTime: '2017-05-04T12:51:15.123Z'}}
      day={3}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});

test('Renders properly', () => {
  const updateMarketStatus = jest.fn();
  const component = shallow(
    <MarketStatusPage
      updateMarketStatus={updateMarketStatus}
      status={{day: 4, open: false, dateTime: '2017-05-05T17:51:15.123Z'}}
      day={4}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});

// MarketStatusPage.propTypes = {
//   status: PropTypes.object,
//   updateMarketStatus: PropTypes.func,
//   day: PropTypes.number
// };
