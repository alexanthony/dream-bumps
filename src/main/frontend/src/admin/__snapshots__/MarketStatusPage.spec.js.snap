exports[`test Renders properly 1`] = `
<Grid>
  <Cell
    col={6}>
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardText>
        <h3>
          The market is currently 
        </h3>
        <h1>
          Open
        </h1>
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Close Market"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false} />
      </CardText>
    </Card>
  </Cell>
  <Cell
    col={6}>
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardText>
        <h3>
          The market is currently showing
        </h3>
        <h1>
          Start Order for Day 4
        </h1>
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Next Day"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false} />
        <p>
          This cannot be reversed easily. This will also crystallise the dividends for crews. Make sure you have entered all bumps before this.
        </p>
      </CardText>
    </Card>
  </Cell>
  <Cell
    col={6}>
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardText>
        <p>
          Market status last updated: 
          <Timestamp
            format="ago"
            includeDay={false}
            offset={0}
            time="2017-05-04T12:51:15.123Z" />
        </p>
      </CardText>
    </Card>
  </Cell>
</Grid>
`;

exports[`test Renders properly 2`] = `
<Grid>
  <Cell
    col={6}>
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardText>
        <h3>
          The market is currently 
        </h3>
        <h1>
          Closed
        </h1>
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Open Market"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false} />
      </CardText>
    </Card>
  </Cell>
  <Cell
    col={6}>
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardText>
        <h3>
          The market is currently showing
        </h3>
        <h1>
          Final Finish Order
        </h1>
        <RaisedButton
          disabled={true}
          fullWidth={false}
          label="Next Day"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false} />
        <p>
          This cannot be reversed easily. This will also crystallise the dividends for crews. Make sure you have entered all bumps before this.
        </p>
      </CardText>
    </Card>
  </Cell>
  <Cell
    col={6}>
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardText>
        <p>
          Market status last updated: 
          <Timestamp
            format="ago"
            includeDay={false}
            offset={0}
            time="2017-05-05T17:51:15.123Z" />
        </p>
      </CardText>
    </Card>
  </Cell>
</Grid>
`;
